Lib/datetime.py

```diff
@@ -237,6 +237,89 @@ def _wrap_strftime(object, format, timetuple):
     newformat = "".join(newformat)
     return _time.strftime(newformat, timetuple)
 
+# Helpers for parsing the result of isoformat()
+def _parse_isoformat_date(dtstr):
+    # It is assumed that this function will only be called with a
+    # string of length exactly 10, and (though this is not used) ASCII-only
+    year = int(dtstr[0:4])
+    if dtstr[4] != '-':
+        raise ValueError('Invalid date separator: %s' % dtstr[4])
+
+    month = int(dtstr[5:7])
+
+    if dtstr[7] != '-':
+        raise ValueError('Invalid date separator')
+
+    day = int(dtstr[8:10])
+
+    return [year, month, day]
+
+def _parse_isoformat_time(tstr):
+    len_str = len(tstr)
+    if len_str < 2:
+        raise ValueError('Isoformat time too short')
+
+    time_comps = [0, 0, 0, 0, None]
+    pos = 0
+    comp = 0
+    while True:
+        if (len_str - pos) < 2:
+            raise ValueError('Incomplete time component')
+
+        time_comps[comp] = int(tstr[pos:pos+2])
+
+        pos += 2
+        comp += 1
+        next_char = tstr[pos:pos+1]
+        if comp > 2 or next_char in '-+':
+            break
+
+        if next_char != ':':
+            raise ValueError('Invalid time separator')
+        if pos >= len_str:
+            break
```

Comment:
In retrospect, I think maybe it's logically impossible to hit this branch, since we only get here if `tstr[pos:pos+1] == ':'`. I'll remove this in the next iteration.